pkg/controller/certificates/trigger/trigger_controller.go

```diff
@@ -147,9 +147,25 @@ func (c *controller) ProcessItem(ctx context.Context, key string) error {
 		return nil
 	}
 
-	// ensure a resync is scheduled in the future so that we re-check
-	// Certificate resources and trigger them near expiry time
-	c.scheduleRecheckOfCertificateIfRequired(log, key, crt.Status.RenewalTime)
+	// check if we have had a recent failure, and if so do not trigger a
+	// reissue immediately
+	if crt.Status.LastFailureTime != nil {
+		// the amount of time after the LastFailureTime before the request should be retried
+		const retryAfterLastFailure = time.Hour
+		now := c.clock.Now()
+		retryAfter := crt.Status.LastFailureTime.Add(retryAfterLastFailure)
+		if now.Before(retryAfter) {
+			log.Info("Not re-issuing certificate as an attempt has been made in the last hour", "retry_after", retryAfter)
+			c.scheduleRecheckOfCertificateIfRequired(log, key, retryAfter)
```

Comment:
Edge case of `now` being before `retryAfter` in if check above but not anymore in `scheduleRecheckOfCertificateIfRequired`
(@munnerz mentioned this in the stand-up, writing it down just in case)

Reply:
I've adapted this method to accept a duration instead of a time, which avoids it needing to reference `now` anywhere (thus avoiding this problem) 😄